### DISCUSSION TITLE
Claude Desktop App with HTTP-with-SSE-transport?

### DISCUSSION THREAD

**Discussion by dforsber** [modelcontextprotocol/.github#16] [Q&A] [answered]
Configuring the MCP servers into Claude Desktop App currently only show how to add the stdio protocol versions. 
==> How do I add MCP server using HTTP with SSE transport? Or is it not yet supported?

**Comment by dsp-ant** [comment]
This is not supported at the moment.

**Reply by shenli** [reply]
May I know if this is on the roadmap?

**Reply by tadasant** [reply]
Here is a great workaround while we wait for official support: https://developers.cloudflare.com/agents/guides/remote-mcp-server/#connect-your-remote-mcp-server-to-claude-and-other-mcp-clients-via-a-local-proxy

**Reply by tadasant** [reply]
Remote MCP is now supported (for certain plans): https://support.anthropic.com/en/articles/11175166-about-custom-integrations-using-remote-mcp

**Reply by julien-c** [reply]
do you have it in the Claude Max plan? or is it also geographically-restricted?

**Reply by jiri-prokop-pb** [reply]
> Remote MCP is now supported (for certain plans): https://support.anthropic.com/en/articles/11175166-about-custom-integrations-using-remote-mcp

Not really. It's quite different implementation. I would expect to be able to use local SSE too, with the regular MCP configuration, similar to other tools like Cursor or VSCode.

**Reply by tadasant** [reply]
Most of us responding here in the MCP community are not affiliated with Anthropic - I encourage you all to use [Anthropic's official support channels](https://support.anthropic.com/en/articles/11175166-getting-started-with-custom-integrations-using-remote-mcp) if you have feedback on their products.

**Comment by dforsber** [comment]
I just wrote a stdio to http sse gateway with example MCP server and client as well. The server exposes DuckDB database and allows Claude to run queries on it.
https://github.com/boilingdata/mcp-server-and-gw

**Reply by shenli** [reply]
It works for me. Thanks!

**Comment by hasani114** [comment]
Is there any update on this? I am able to use this on LibreChat and Cursor now but it doesn't work on Claude Desktop app

**Comment by SamFold** [comment]
Are SSE Servers supported with Claude Desktop yet? If so, what does the configuration look like?

**Reply by QuantGeekDev** [reply]
I don't think they are unfortunately, I'm looking for the same

**Comment by dforsber** [comment]
Updated the README in the example stdio to HTTP SSE bridge:
https://github.com/boilingdata/mcp-server-and-gw

Shouldn't be too hard to convert to python if you prefer that. Naturally, you can run your server in Python if you like.

**Comment by dacay** [comment]
**Could you please at least state whether this is on the roadmap or not?**

I am working on a somewhat complex MCP server and currently, it only supports HTTP SSE due to some OS-level permission issues. If it was an Open Source project, I'd love to contribute as well :)

**Comment by dasoncheng** [comment]
Claude Desktop App with SSEServerTransport, what does the configuration look like?

**Reply by QuantGeekDev** [reply]
Like nothing. Anthropic hasn't really added any of that stuff yet. Just the stdio tool calling. Hope it helps

**Reply by shushui** [reply]
> Like nothing. Anthropic hasn't really added any of that stuff yet. Just the stdio tool calling. Hope it helps

I'm not sure if I'm shocked or that it is obvious for me.
I developed MCP server with the problematic SSE definition, feeling Anthropic didn't really got into supporting any remote MCP servers properly. I bumped into many issues and overcome them, just to find out Claude app doesn't even support this.
Since Claude app wasn't my go-to client, it is fine, but it feels lame.

**Comment by HectorDavila** [comment]
This worked for me: https://developers.cloudflare.com/agents/guides/test-remote-mcp-server/#connect-your-remote-mcp-server-to-claude-desktop-via-a-local-proxy

**Comment by CyrilPeponnet** [comment]
⚠️ self promotion

We made Minibridge to support that use case for free and more https://github.com/acuvity/minibridge

**Comment by mzhadigerov** [comment]
any updates on this?

**Comment by adstastic** [comment]
It's working from Claude Code with FastMCP's SSE mode: 
```
mcp.run(transport="sse", host="127.0.0.1", port=8000)
```

**Reply by pannous** [reply]
But how do you specify sse in claude_desktop_config.json ?

**Comment by renato-umeton** [comment]
Any update on this?

**Comment by pgangwani** [comment]
WHy is MCP creator so behind for this support?

**Comment by tadasant** [comment, marked as answer]
Remote MCP support in Claude is now documented here: https://support.anthropic.com/en/articles/11175166-about-custom-integrations-using-remote-mcp

**Reply by jiri-prokop-pb** [reply]
Yes, but it sucks. I would expect support similar to other tools where you can have local MCPs that run as servers.

As I mentioned here: https://github.com/orgs/modelcontextprotocol/discussions/16#discussioncomment-13035353

**Reply by tadasant** [reply]
Most of us responding here in the MCP community are not affiliated with Anthropic - I encourage you all to use [Anthropic's official support channels](https://support.anthropic.com/en/articles/11175166-getting-started-with-custom-integrations-using-remote-mcp) if you have feedback on their products.

**Comment by DePasqualeOrg** [comment]
"Custom connectors" don't work on localhost with http, since Claude Desktop requires that the URL start with https. Could you please allow http on localhost so that people can easily run their own servers?

**Reply by eng-marani** [reply]
I'm facing the same exact issue.
I was looking to use Claude desktop to test my own MCP server running locally on Http (not https) but as far as i know the "command" parameter is mandatory, and i've found no way of adding my MCP server as local tools.
It seem that someone have succeed using a "local proxy", do anyone know how to setup one of these ?

**Reply by dacay** [reply]
> I'm facing the same exact issue.
> 
> I was looking to use Claude desktop to test my own MCP server running locally on Http (not https) but as far as i know the "command" parameter is mandatory, and i've found no way of adding my MCP server as local tools.
> 
> It seem that someone have succeed using a "local proxy", do anyone know how to setup one of these ?

Using ngrok should work. It is much easier to set up compared to a reverse proxy. That would require something like Nginx and configuration to use a self-signed certificate.

**Comment by kerryjiang** [comment]
Yes, same problem here. But According to Claude's doc, http is supported.
https://code.claude.com/docs/en/mcp

**Reply by DePasqualeOrg** [reply]
It's true, they have examples with `http://localhost` on that page, but it doesn't work in Claude Desktop. @bcherny, @ThariqS, any chance you could pass this on to someone to get this fixed?

**Comment by eng-marani** [comment]
In order to test your http MCP server using claude desktop, you need to use this configuration, that is unnecessary if you have the "pro" paied version, but mandatory with the free version, as far as the "command" keyword is mandatory.

Example of **claude_desktop_config.json**:

```
{
  "mcpServers": {
    "MyMCP-server": {
      "command": "npx",
      "args": ["mcp-remote", "http://localhost:3353/sse"]
    }
  }
}

```
Use your MCP url address with proper port as needed.
Hope this helps,
Paolo

**Reply by swiru95** [reply]
ok but what about kind of custom headers and auth requirements?

**Comment by DePasqualeOrg** [comment]
I created this package to bridge local MCP servers running on HTTP to stdio for Claude Desktop, which rejects URLs starting with `http://` even on localhost.

https://github.com/DePasqualeOrg/mcp-http-stdio-bridge

You can use it like this in `claude_desktop_config.json`:

```json
{
  "mcpServers": {
    "My Server": {
      "command": "npx",
      "args": ["-y", "@depasquale/mcp-http-stdio-bridge", "--url", "http://localhost:8080/mcp"]
    }
  }
}
```

**Reply by swiru95** [reply]
What if I need to provide headers, auth etc?:D
